packages/sdk/src/oracle/services/erc20.ts

```diff
@@ -0,0 +1,43 @@
+import { erc20 } from "../../clients";
+import Multicall2 from "../../multicall2";
+import { BatchReadWithErrors, BatchReadWithErrorsType, Calls } from "../../utils";
+import { BigNumber, Provider } from "../types/ethers";
+
+const batchProps: Calls = [["symbol"], ["name"], ["decimals"], ["totalSupply"]];
+type Props = {
+  symbol: string;
+  name: string;
+  decimals: number;
+  totalSupply: BigNumber;
+};
+export class Erc20 {
+  public contract: erc20.Instance;
```

Comment:
could not find a way to extend from this, due to the factory being used to instantiate in typechain

Reply:
Can you explain further?

Reply:
Ideally i twould be nice to be able to extend the contract class and just add functions to it, but the implementation is required to do this and only the factory is exposed afaik.  Given that, its not clear how you would instantiate "this" with a factory, rather than a constructor.

Reply:
Oh I see. I think it's actually kind of nice to keep the client and contract objects separate. But your explanation totally makes sense.